libdnf/module/module_persistor.hpp

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_PERSISTOR_HPP
+#define LIBDNF_MODULE_MODULE_PERSISTOR_HPP
+
+#include "libdnf/module/module_item_container.hpp"
+#include "libdnf/system/state.hpp"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace libdnf::module {
+
+
+class ModulePersistor {
+public:
+    ModulePersistor(libdnf::system::State & system_state);
+
+    bool add_module();
+    void save_everything_to_system_state();
+
+    const ModuleState & get_state(const std::string & module_name);
```

Comment:
I think it is a getter and it must not change the value of the class members. Please mark it as `const`.
`const ModuleState & get_state(const std::string & module_name) const;`

Also fix other getters.

A `const` version of the `get_runtime_module_state` method will need to be added to make the change possible.